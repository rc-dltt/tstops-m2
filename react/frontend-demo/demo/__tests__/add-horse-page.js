import React from "react";
import { MockedProvider } from '@apollo/client/testing';
import { PaperProvider } from 'react-native-paper';
import AddHorsePage from "../Components/AddHorsePage";
import { allRaceQuery, allHorseQuery } from "../graphql/query"
import { loginMutation } from '../graphql/mutation';
import { fireEvent, cleanup, render } from "@testing-library/react-native";

const mocks = [
    {
        request: {
            query: loginMutation,
            variables: {
                email: 'johndoe@email.com',
                password: 'pAsSWoRd!',
            },
        },
        result: {
            data: {
                login: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6InVzZXIjMSIsInJvbGVzIjpbInVzZXIiLCJhZG1pbiJdLCJpYXQiOjE2OTI2NzE5NTEsImV4cCI6MTY5Mjc1ODM1MSwic3ViIjoidXNlciMxIn0.3CPT3fQqU6fLarIh8y0zFkJD7Xik5UdzBBkapqZdzSM"
            },
        },
    },
    {
        request: {
            query: allRaceQuery,
        },
        result: {
            data: {
                races: {
                    id: '84',
                    no: "#1",
                    startTime: "2023-08-12",
                    venue: "Shat Tin"
                }
            },
        },
    },
    {
        request: {
            query: allHorseQuery,
        },
        result: {
            data: {
                horses: {
                    id: '103',
                    name: "Test Horse",
                    rank: "123"
                }
            },
        },
    },
];

beforeEach(() => {
    jest.useFakeTimers();
});

describe('Add Horse Page', () => {
    it('Snapshot Test', () => {
        const addHorsePage = render(
            <MockedProvider mocks={mocks} addTypename={false} >
                <PaperProvider>
                    <AddHorsePage />
                </PaperProvider>
            </MockedProvider>
        );
        expect(addHorsePage).toMatchSnapshot();
    })
})
