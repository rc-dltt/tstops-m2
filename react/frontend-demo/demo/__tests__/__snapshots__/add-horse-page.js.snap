// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Horse Page Snapshot Test 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      contentInsetAdjustmentBehavior="automatic"
      style={
        {
          "backgroundColor": "#F3F3F3",
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "#FFF",
          }
        }
      >
        <View
          style={
            {
              "marginTop": 32,
              "paddingHorizontal": 24,
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontSize": 24,
                  "fontWeight": "600",
                },
                {
                  "color": "#000",
                },
              ]
            }
          >
            Mutation - Add Horse
          </Text>
          <Text
            style={
              [
                {
                  "fontSize": 18,
                  "fontWeight": "400",
                  "marginTop": 8,
                },
                {
                  "color": "#444",
                },
              ]
            }
          >
            Input Horse Name and Horse Rank to Add Horse.
          </Text>
        </View>
      </View>
      <TextInput
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Enter Horse Name"
        style={
          {
            "borderWidth": 1,
            "height": 40,
            "margin": 12,
            "padding": 10,
          }
        }
        testID="add-horse-name-input"
        value=""
      />
      <TextInput
        keyboardType="default"
        onChangeText={[Function]}
        placeholder="Enter Horse Rank"
        style={
          {
            "borderWidth": 1,
            "height": 40,
            "margin": 12,
            "padding": 10,
          }
        }
        testID="add-horse-rank-input"
        value=""
      />
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "#424d9c",
            "borderRadius": 20,
            "margin": 12,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="add-horse-btn-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "#424d9c",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="add-horse-btn-container"
          title="submit"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                {
                  "borderRadius": 20,
                },
              ]
            }
            testID="add-horse-btn"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  [
                    {
                      "textAlign": "left",
                    },
                    {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    [
                      [
                        {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        {
                          "marginHorizontal": 24,
                          "marginVertical": 10,
                        },
                        undefined,
                        false,
                        {
                          "color": "#fbde2b",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        undefined,
                      ],
                      undefined,
                    ],
                  ]
                }
                testID="add-horse-btn-text"
              >
                <Text>
                  Submit
                </Text>
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
</RNCSafeAreaProvider>
`;
